doc/mgmt/Management Framework.md

```diff
@@ -1922,7 +1925,87 @@ The table-transformer is inherited to all descendant nodes unless another one is
 11. `sonic-ext:get-validate [function]`: 
 A special hook to validate YANG nodes, to populate data read from database, allows developers to instruct Transformer to choose a YANG node among multiple nodes, while constructing the response payload. 
 Typically used to check the “when” condition to validate YANG node among multiple nodes to choose only valid nodes from sibling nodes.
+12. `sonic-ext:get-namespace [function]`:
+Multiple DB extension, populates associated namespaces with the YANG path
+Used by Translib service to get all DB's associated with the YANG path. In case of single DB, this extension shall not be used.
+Namespace yang extension shall allow override capability to allow product specific customizations 
+```Sample annotation
+get-namespace annotation sample for openconfig-optical-amplifier yang support: 
+
+Annotation YANG File:
+sonic-mgmt-common/models/yang/annotations/openconfig-optical-amplifier-annot.yang
+module openconfig-optical-amplifier-annot {
+  .....
+    deviation /oc-opt-amp:optical-amplifier/oc-opt-amp:amplifiers/oc-opt-amp:amplifier {
+      deviate add {
+        sonic-ext:key-transformer "oc_name_key_xfmr";
+        sonic-ext:get-namespace "oc_name_get_namespace_xfmr";
+      }
+    }
+  .....
+    deviation /oc-opt-amp:optical-amplifier/oc-opt-amp:amplifiers/oc-opt-amp:amplifier/oc-opt-amp:state {
+      deviate add {
+        sonic-ext:db-name "STATE_DB";
+        sonic-ext:table-name "AMPLIFIER";
+        sonic-ext:key-transformer "oc_name_key_xfmr";
+        sonic-ext:get-namespace "oc_name_get_namespace_xfmr";
+      }
+    }
+  .....
+    deviation /oc-opt-amp:optical-amplifier/oc-opt-amp:supervisory-channels/oc-opt-amp:supervisory-channel {
+      deviate add {
+        sonic-ext:key-transformer "osc_key_xfmr";
+        sonic-ext:get-namespace "oc_name_get_namespace_xfmr";
+      }
+    }
+  .....
+    deviation /oc-opt-amp:optical-amplifier/oc-opt-amp:supervisory-channels/oc-opt-amp:supervisory-channel/oc-opt-amp:state {
+      deviate add {
+        sonic-ext:db-name "STATE_DB";
+        sonic-ext:table-name "OSC";
+        sonic-ext:key-transformer "osc_key_xfmr";
+        sonic-ext:get-namespace "oc_name_get_namespace_xfmr";
```

Comment:
If name-space callback is going to be inherited why is the child yang node again annotated with name-space callback and the same one as that of the parent?

Reply:
> If name-space callback is going to be inherited why is the child yang node again annotated with name-space callback and the same one as that of the parent?

Yes, will remove the name-space annotation for child node